hoomd/mpcd/CollisionMethod.cc

```diff
@@ -48,15 +48,272 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
     // sync the embedded group
     m_cl->setEmbeddedGroup(m_embed_group);
 
+    // check for collision warnings
+    checkCollisionWarnings(timestep);
+
     // set random grid shift
     m_cl->drawGridShift(timestep);
 
     // update cell list
     m_cl->compute(timestep);
 
+    // create auxillary array for rigid bodies
+    if (m_embed_group)
+        {
+        // resize initial velocity array
+        const unsigned int num_group = m_embed_group->getNumMembers();
+        if (num_group > m_initial_velocity.getNumElements())
+            {
+            GPUArray<Scalar4> initial_velocity(num_group, m_exec_conf);
+            m_initial_velocity.swap(initial_velocity);
+            }
+
+        const unsigned int num_total = m_pdata->getN();
+        if (num_total > m_linmom_accum.getNumElements())
+            {
+            GPUArray<Scalar3> linmom_accum(num_total, m_exec_conf);
+            m_linmom_accum.swap(linmom_accum);
+            GPUArray<Scalar4> angmom_accum(num_total, m_exec_conf);
+            m_angmom_accum.swap(angmom_accum);
+            }
+        beginRigidBodyCollision(timestep);
+        }
+
+    // apply collisions
     rule(timestep);
+
+    // apply collisions to rigid bodies
+    if (m_embed_group)
+        {
+        accumulateRigidBodyMomenta(timestep);
+        finishRigidBodyCollision(timestep);
```

Comment:
Making a note that we will also want to update the constituent particle velocities. For that, we'll need to make sure the collision method has a pointer to the rigid body constraint object. The `Integrator` will need to forward this. That's nice because then you can also use that pointer to check whether the rigid body steps are necessary (instead of only testing for `m_embed_group`).

We can discuss how to do that if you are unsure.

Reply:
I figured out how to do it.